pkg/kubelet/status/state/state_mem.go

```diff
@@ -74,12 +74,19 @@ func (s *stateMemory) SetContainerResourceAllocation(podUID string, containerNam
 	return nil
 }
 
-func (s *stateMemory) SetPodResourceAllocation(a PodResourceAllocation) error {
+func (s *stateMemory) SetPodResourceAllocation(podUID string, alloc map[string]v1.ResourceRequirements) error {
 	s.Lock()
 	defer s.Unlock()
 
-	s.podAllocation = a.Clone()
-	klog.V(3).InfoS("Updated pod resource allocation", "allocation", a)
+	if _, ok := s.podAllocation[podUID]; !ok {
+		s.podAllocation[podUID] = make(map[string]v1.ResourceRequirements)
+	}
+
+	for containerName, containerAlloc := range alloc {
+		s.podAllocation[podUID][containerName] = containerAlloc
+	}
```

Comment:
can't we just do
```go
s.podAllocation[podUID] = alloc
```
here? Any reason we need to deep-copy the alloc map?

Reply:
I think this comment still stands? 

```suggestion
	s.podAllocation[podUID] = alloc
```

Reply:
perfect, adjusted!